src/annotation/decorator_utils.ts

```diff
@@ -1,68 +1,97 @@
 import * as ERROR_MSGS from "../constants/error_msgs";
 import * as METADATA_KEY from "../constants/metadata_keys";
 import { interfaces } from "../interfaces/interfaces";
+import { getFirstArrayDuplicate } from "../utils/js";
 
 function tagParameter(
     annotationTarget: any,
     propertyName: string,
```

Comment:
this is now `string | undefined` (we want to set strict mode soon)

Reply:
Have gone for `string | symbol | undefined` to match the typescript decorator definition.

> declare type ParameterDecorator = (target: Object, propertyKey: string | symbol, parameterIndex: number) => void;

Although when would it be a symbol ?

Reply:
Also had to cast to unknown first when calling as cannot add undefined to targetKey

https://github.com/microsoft/TypeScript/issues/33260